### PR TITLE
workflow: dogfood & tidy

### PR DESCRIPTION
**Pull request by casperdcl**
- minify/tidy/lint workflows
- bump actions versions
- CML dogfooding
- make releases `--draft`

### CODE REVIEW

File: .github/workflows/images.yml [hunk 1]
```diff
@@ -1,9 +1,7 @@
 on:
   workflow_call:
     inputs:
-      test:
-        required: false
-        type: boolean
+      test: { required: false, type: boolean }
```
Comment (dacbd) [inline .github/workflows/images.yml:4]: 👎

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -8,30 +8,31 @@ on:
         description: Bump version number
         options: [major, minor, patch]
         default: patch
-  pull_request:
-    types: [closed]
+  pull_request: { types: [closed] }
```
Comment (dacbd) [inline .github/workflows/release.yml:11]: 👎

File: .github/workflows/test-deploy.yml [hunk 3]
```diff
@@ -1,9 +1,7 @@
 name: Test & Deploy
 on:
-  schedule:
-    - cron: '0 8 * * 1' # M H d m w (Mondays at 8:00)
-  release:
-    types: [created]
+  schedule: [{ cron: '0 8 * * 1' }] # M H d m w (Mondays at 8:00)
+  release: { types: [created] }
```
Comment (dacbd) [inline .github/workflows/test-deploy.yml:4]: 👎

File: .github/workflows/trigger-external.yml [hunk 4]
```diff
@@ -6,25 +6,19 @@ on:
...
         type: string
         required: true
-  release:
-    types:
-      - published
-  pull_request:
-    branches:
-      - master
+  release: { types: [published] }
+  pull_request: { branches: [master] }
```
Comment (dacbd) [inline .github/workflows/trigger-external.yml:10]: 👎

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -8,30 +8,31 @@ on:
...
     if: github.event.pull_request.merged && startsWith(github.head_ref, 'bump/')
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - run:
-          gh release create --generate-notes {--title=CML\ ,}$(basename ${{
-          github.head_ref }})
+          gh release create --draft --generate-notes {--title=CML\ ,}$(basename
+          ${{ github.head_ref }})
```
Comment (dacbd) [inline .github/workflows/release.yml:36]: do we need to wrap this line?

```suggestion
          gh release create --draft --generate-notes {--title=CML\ ,}$(basename ${{ github.head_ref }})
```

**Review by dacbd — COMMENTED** [review]
I am personally not a fan of the `{}` syntax in yaml, additionally its even less commonly seen in the scope of GitHub Actions.

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -8,30 +8,31 @@ on:
...
+      - name: Bump version
+        run: |
+          cml ci
           git checkout -b bump/$(npm version ${{ github.event.inputs.bump }})
+      - name: Create PR
+        run: |
           git push --set-upstream origin --follow-tags HEAD
-          gh pr create --title "Bump version to $(git describe --tags)" --body "Approve me 🤖"
-          gh pr merge --squash --auto
+          cml pr create --title "Bump version to $(git describe --tags)" --body "Approve me 🤖" --squash
```
Comment (0x2b3bfa0) [inline .github/workflows/release.yml:26]: Are you sure that using CML to release CML is a good idea, taking into account that forges' API can break and cause a catch-22 situation? 🤔

File: .github/workflows/release.yml (hunk 5 above)
Comment (0x2b3bfa0) [reply .github/workflows/release.yml:36]: Probably not. It's a consequence of Codespaces' editor settings. 🙈

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -8,30 +8,31 @@ on:
...
   release:
     if: github.event.pull_request.merged && startsWith(github.head_ref, 'bump/')
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - run:
-          gh release create --generate-notes {--title=CML\ ,}$(basename ${{
-          github.head_ref }})
+          gh release create --draft --generate-notes {--title=CML\ ,}$(basename
```
Comment (0x2b3bfa0) [inline .github/workflows/release.yml:35]: <sup>[justification needed]</sup>

**Comment by 0x2b3bfa0** [comment]
> _I am personally not a fan of the {} syntax_

I wholeheartedly agree: it's `:purugly:`

File: .github/workflows/release.yml (hunk 6 above)
Comment (dacbd) [reply .github/workflows/release.yml:26]: instead of `setup-cml` `npm install -g .`

Comment (0x2b3bfa0) [reply .github/workflows/release.yml:26]: Clever! :+1:

File: .github/workflows/images.yml (hunk 1 above)
Comment (casperdcl) [reply .github/workflows/images.yml:4]: actually, is this `test` input even used in this workflow @0x2b3bfa0 ?

File: .github/workflows/release.yml (hunk 6 above)
Comment (casperdcl) [reply .github/workflows/release.yml]: > both CML and forges' API can break

exactly, in which case the release will fail noisily.

File: .github/workflows/release.yml (hunk 7 above)
Comment (casperdcl) [reply .github/workflows/release.yml:35]: human notes currently are far superior to the auto-generated ones

**Comment by casperdcl** [comment]
> personally not a fan of the `{}` syntax

`-Wnot-using-a-legitimate-feature`? This syntax exists for precisely this use case (short one-liner unlikely to change).

> less commonly seen in the scope of GitHub Actions

[fallacious, scanadlous!](https://en.wikipedia.org/wiki/Argumentum_ad_populum)

File: .github/workflows/release.yml (hunk 5 above)
Comment (casperdcl) [reply .github/workflows/release.yml:36]: the pre-commit hooks/linter insists on this wrapping. Would be happy if anyone wants to mod `.eslintrc.js`/`prettier.config.js/`.restyled.yaml`...

Comment (casperdcl) [reply .github/workflows/release.yml:36]: btw also if you run `npm run lintfix` locally there are loads of errors... worth a follow-up issue?

File: .github/workflows/release.yml (hunk 7 above)
Comment (0x2b3bfa0) [reply .github/workflows/release.yml:35]: But you can edit them, regardless of the `--draft` option 🤔

**Comment by 0x2b3bfa0** [comment]
> _`-Wnot-using-a-legitimate-feature`_

For the same reason some code files contain line breaks, even if they don't have to: readability.

File: .github/workflows/release.yml (hunk 5 above)
Comment (0x2b3bfa0) [reply .github/workflows/release.yml:36]: Both of your last comments are worth a follow-up issue. I'm not a fan of that wrapping either.

File: .github/workflows/release.yml (hunk 7 above)
Comment (casperdcl) [reply .github/workflows/release.yml:35]: subscribers will get notifications pre-edit not post-edit. This is v. bad.

**Comment by casperdcl** [comment]
> For the same reason some code files contain line breaks

[Still not right](https://simple.wikipedia.org/wiki/False_analogy) - an accurate analogy would be refusing to use line breaks. Refusing to use a feature regardless of scenario is inexplicable.

> readability

yes, maximising that here. Also not using `{}` where it doesn't make sense/harms readability. Also the linter is v. happy with the proposed use of `{}` :)

File: .github/workflows/images.yml (hunk 1 above)
Comment (0x2b3bfa0) [reply .github/workflows/images.yml:4]: It's not 🙈

File: .github/workflows/release.yml (hunk 7 above)
Comment (0x2b3bfa0) [reply .github/workflows/release.yml:35]: Good point

File: .github/workflows/images.yml (hunk 1 above)
Comment (casperdcl) [reply .github/workflows/images.yml:4]:
```suggestion
```

**Comment by tasdomas** [comment]
Re `{}` - that syntax has limited (very limited) uses in yaml, imho. Linters are not raising issues because valid json is valid yaml.

File: .github/workflows/release.yml (hunk 2 above)
Comment (casperdcl) [reply .github/workflows/release.yml]: reverted

File: .github/workflows/trigger-external.yml (hunk 4 above)
Comment (casperdcl) [reply .github/workflows/trigger-external.yml]: reverted

File: .github/workflows/test-deploy.yml (hunk 3 above)
Comment (casperdcl) [reply .github/workflows/test-deploy.yml]: reverted

**Comment by casperdcl** [comment]
[dropped {} by popular demand](https://github.com/iterative/cml/pull/1165/commits/96cbc0eb45cf1ef7a7a900021bf543de2715d9c5)

Also pushed [4b933a6](https://github.com/iterative/cml/pull/1165/commits/4b933a678f13a9f7ea06284e2d6a5846cd67890b) after a thorough review (related to https://github.com/iterative/cml.dev/pull/324) - please do check /CC @0x2b3bfa0

File: .github/workflows/release.yml (hunk 5 above)
Comment (casperdcl) [reply .github/workflows/release.yml:37]: done: #1177, #1178

**Pull request merged**